### PR TITLE
Add some auth handling for etcd

### PR DESCRIPTION
**Pull request by techhat**
Ping @meggiebot.

### CODE REVIEW

File: salt/returners/etcd_return.py [hunk 1]
```diff
@@ -69,11 +93,12 @@ def __virtual__():
...
 
 
-def _get_conn(opts):
+def _get_conn(opts, profile=None):
     '''
     Establish a connection to etcd
     '''
-    profile = opts.get('etcd.returner', None)
+    if profile is None:
+        profile = opts.get('etcd.returner', None)
```
Comment (cachedout) [inline salt/returners/etcd_return.py]: The `None` is redundant here, FWIW. It's the default from the `get()` for a failed lookup.

**Comment by techhat** [comment]
Fixed.

File: salt/returners/etcd_return.py (hunk 1 above)
Comment (techhat) [inline salt/returners/etcd_return.py]: Fixed.

**Comment by cachedout** [comment]
This should be good to merge once the tests pass.

**Pull request merged**